opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -208,8 +209,38 @@ def set_attribute(self, key: str, value: types.AttributeValue) -> None:
         if has_ended:
             logger.warning("Setting attribute on ended span.")
             return
+
+        if not isinstance(value, (int, float, bool, str, list, tuple)):
+            logger.warning("invalid type for attribute value")
+            return
+        if isinstance(value, (list, tuple)) and len(value) > 0:
+            return_code = self._check_sequence(value)
+            if return_code is not None:
+                logger.warning("%s in attribute value sequence", return_code)
+                return
+
         self.attributes[key] = value
 
+    @staticmethod
+    def _check_sequence(sequence: (list, tuple)) -> Optional[str]:
+        """
+        Checks if sequence items are valid and identical
+        """
+        first_element_type = type(sequence[0])
+
+        if issubclass(type(sequence[0]), Number):
+            first_element_type = Number
+
+        for element in sequence:
+
+            if not isinstance(element, (bool, str, Number)):
```

Comment:
should this tuple be moved into a constant in the module? This set should be shared with the tuple on line 213 (could modify line 213 to be that list + list, tuple:

`VALID_ATTRIBUTE_TYPES + (list, tuple)`

Reply:
I agree this should be a constant. But do you think it might be misleading to have the constant `VALID_ATTRIBUTE_TYPES` without `list, tuple` included, since they are actually valid?  

Maybe we could define `VALID_ATTRIBUTE_SEQUENCE_TYPES` and `VALID_ATTRIBUTE_NON_SEQUENCE_TYPES`, then define `VALID_ATTRIBUTE_TYPES` as the union of the two. What do you think?